### ISSUE TITLE
[JENKINS-3121] Hudson returns http 405 when parameterized option is checked

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#2304]
<p>When the "parameterized" option is checked within a build, Hudson is returning a<br/>
HTTP 405.</p>

<p>This was using Hudson 1.285</p>

<p>This is a problem when using Hudson behind a proxy as the proxy will return only<br/>
the HTTP 405 response code.</p>

<p>Edit 2013/05/03: Some reverse proxies generate their own error pages, so Jenkins should not rely on the HTML body in the 405 response being presented to the user. This body is currently used to specify the parameters to use on a parameterized build.</p>

<p>$ curl -v <a href="http://cu009.cubit.ibitdev.com:8080/job/test_params/build?delay=0sec" class="external-link" target="_blank" rel="nofollow noopener">http://cu009.cubit.ibitdev.com:8080/job/test_params/build?delay=0sec</a></p>
<ul>
	<li>About to connect() to cu009.cubit.ibitdev.com port 8080 (#0)</li>
	<li>Trying 208.87.8.211... connected</li>
	<li>Connected to cu009.cubit.ibitdev.com (208.87.8.211) port 8080 (#0)<br/>
> GET /job/test_params/build?delay=0sec HTTP/1.1<br/>
> User-Agent: curl/7.16.3 (i686-pc-cygwin) libcurl/7.16.3 OpenSSL/0.9.8i zlib/1.<br/>
2.3 libssh2/0.15-CVS<br/>
> Host: cu009.cubit.ibitdev.com:8080<br/>
> Accept: <b>/</b><br/>
><br/>
< HTTP/1.1 405 Method Not Allowed<br/>
< Server: Winstone Servlet Engine v0.9.10<br/>
< Expires: 0<br/>
< Content-Type: text/html;charset=UTF-8<br/>
< Connection: Close<br/>
< Date: Mon, 23 Feb 2009 20:08:45 GMT<br/>
< X-Powered-By: Servlet/2.5 (Winstone/0.9.10)<br/>
< Set-Cookie: JSESSIONID=56cb51b13ed79bbc1c91c45fb397022f; Path=/<br/>
<<br/>
<html><head></li>
</ul>

<p>    <title>Hudson</title><link type="text/css" rel="stylesheet" href="/static/0d<br/>
946710/css/style.css"></link><link type="text/css" rel="stylesheet" href="/stati<br/>
c/0d946710/css/color.css"></link><link type="image/vnd.microsoft.icon" rel="shor<br/>
tcut icon" href="/static/0d946710/favicon.ico"></link><script>var isRunAsTest=fa<br/>
lse;</script><script src="/static/0d946710/scripts/prototype.js" type="text/java<br/>
script"></script><script src="/static/0d946710/scripts/behavior.js" type="text/j<br/>
avascript"></script><script src="/static/0d946710/scripts/sortable.js" type="tex<br/>
t/javascript"></script><script src="/static/0d946710/scripts/yui/yahoo/yahoo-min<br/>
.js"></script><script src="/static/0d946710/scripts/yui/dom/dom-min.js"></script<br/>
><script src="/static/0d946710/scripts/yui/event/event-min.js"></script><script<br/>
src="/static/0d946710/scripts/yui/animation/animation-min.js"></script><script s<br/>
rc="/static/0d946710/scripts/yui/container/container-min.js"></script><script sr<br/>
c="/static/0d946710/scripts/yui/connection/connection-min.js"></script><script s<br/>
rc="/static/0d946710/scripts/yui/autocomplete/autocomplete-min.js"></script><scr<br/>
ipt src="/static/0d946710/scripts/yui/menu/menu-min.js"></script><script src="/s<br/>
tatic/0d946710/scripts/yui/element/element-beta-min.js"></script><script src="/s<br/>
tatic/0d946710/scripts/yui/button/button-min.js"></script><script src="/static/0<br/>
d946710/scripts/yui/dragdrop/dragdrop-min.js"></script><script src="/static/0d94<br/>
6710/scripts/hudson-behavior.js" type="text/javascript"></script><link type="tex<br/>
t/css" rel="stylesheet" href="/static/0d946710/scripts/yui/container/assets/cont<br/>
ainer.css"></link><link type="text/css" rel="stylesheet" href="/static/0d946710/<br/>
scripts/yui/assets/skins/sam/skin.css"></link><link type="text/css" rel="stylesh<br/>
eet" href="/static/0d946710/scripts/yui/button/assets/skins/sam/button.css"></li<br/>
nk><link type="text/css" rel="stylesheet" href="/static/0d946710/scripts/yui/men<br/>
u/assets/skins/sam/menu.css"></link><link type="application/opensearchdescriptio<br/>
n+xml" rel="search" title="Hudson" href="/opensearch.xml"></link><meta content="<br/>
INDEX,NOFOLLOW" name="ROBOTS"></meta><link type="application/rss+xml" rel="alter<br/>
nate" title="Hudson:test_params (all builds)" href="rssAll"></link><link type="a<br/>
pplication/rss+xml" rel="alternate" title="Hudson:test_params (all builds) (RSS<br/>
2.0)" href="rssAll?flavor=rss20"></link><link type="application/rss+xml" rel="al<br/>
ternate" title="Hudson:test_params (failed builds)" href="rssFailed"></link><lin<br/>
k type="application/rss+xml" rel="alternate" title="Hudson:test_params (failed b<br/>
uilds) (RSS 2.0)" href="rssFailed?flavor=rss20"></link><link type="application/r<br/>
ss+xml" rel="alternate" title="Hudson:test_params (changelog)" href="rssChangelo<br/>
g"></link><link type="application/rss+xml" rel="alternate" title="Hudson:test_pa<br/>
rams (changelog) (RSS 2.0)" href="rssChangelog?flavor=rss20"></link></head><body<br/>
 class="yui-skin-sam"><a class="skiplink" href="#skip2content">Skip to content</<br/>
a><table border="0" width="100%" cellpadding="0" cellspacing="0" id="header"><tr<br/>
><td id="top-panel" colspan="2"><table border="0" width="100%" cellpadding="0" c<br/>
ellspacing="0"><tr><td style="font-weight:bold; font-size: 2em;"><a href="/"><im<br/>
g src="/static/0d946710/images/title.png" alt="title"></img></a></td><td style="<br/>
vertical-align: middle; text-align: right; padding-right: 1em;"><form style="pos<br/>
ition:relative;" class="no-json" action="/job/test_params/search/" method="get"<br/>
name="search"><div id="search-box-minWidth"></div><div id="search-box-sizer"></d<br/>
iv><div id="searchform"><input class="has-default-text defaulted" value="search"<br/>
 id="search-box" name="q"></input>A <a href="http://hudson.gotdns.com/wiki/displ<br/>
ay/HUDSON/Search+Box"><img src="/static/0d946710/images/16x16/help.png" alt="hel<br/>
p for search"></img></a><div id="search-box-completion"></div><script>createSear<br/>
chBox("/job/test_params/search/");</script></div></form></td><td id="login-field<br/>
"><span>A <a style="color:inherit" href="/login?from=/job/test_params/build "><b<br/>
>login</b></a></p>
<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
</tr>
</tbody></table>
</div>

<p>    <a href="/signup"><b>sign up</b></a></span></td></tr></table></td></tr><tr i<br/>
d="top-nav"><td id="left-top-nav"><a href="/">Hudson</a> A» <a href="/job/test_p<br/>
arams/">test_params</a></td><td id="right-top-nav"><span class="smallfont"><a hr<br/>
ef="?auto_refresh=true">ENABLE AUTO REFRESH</a></span></td></tr></table><table s<br/>
tyle="background-image: url(/static/0d946710/images/hudson.png);<br/>
background-repeat: no-repeat; background-position: bottom left;" border="0" widt<br/>
h="100%" id="main-table" height="70%"><tr><td width="20%" id="side-panel"><div s<br/>
tyle="margin-left: 20px; margin-right: 20px; border-top: 1px solid #bbb"></div><<br/>
div id="navigation"><div id="tasks"><div class="task"><a href="/"><img src="/sta<br/>
tic/0d946710/images/24x24/up.gif" style="margin: 2px;" width="24" alt="" height=<br/>
"24"></img></a>A <a href="/">Back to Dashboard</a></div><div class="task"><a hre<br/>
f="/job/test_params/"><img src="/static/0d946710/images/24x24/search.gif" style=<br/>
"margin: 2px;" width="24" alt="" height="24"></img></a>A <a href="/job/test_para<br/>
ms/">Status</a></div><div class="task"><a href="/job/test_params/changes"><img s<br/>
rc="/static/0d946710/images/24x24/notepad.gif" style="margin: 2px;" width="24" a<br/>
lt="" height="24"></img></a>A <a href="/job/test_params/changes">Changes</a></di<br/>
v><div class="task"><a href="/job/test_params/ws/"><img src="/static/0d946710/im<br/>
ages/24x24/folder.gif" style="margin: 2px;" width="24" alt="" height="24"></img><br/>
</a>A <a href="/job/test_params/ws/">Workspace</a></div><div class="task"><a hre<br/>
f="/job/test_params/build?delay=0sec"><img src="/static/0d946710/images/24x24/cl<br/>
ock.gif" style="margin: 2px;" width="24" alt="" height="24"></img></a>A <a href=<br/>
"/job/test_params/build?delay=0sec">Build Now</a></div><script><br/>
    function build(a) {<br/>
      new Ajax.Request(a.href,</p>
{method:"post"}
<p>);<br/>
      hoverNotification('Build scheduled',a.parentNode);<br/>
      return false;<br/>
    }<br/>
  </script><div class="task"><a href="/job/test_params/delete"><img src=<br/>
"/static/0d946710/images/24x24/edit-delete.gif" style="margin: 2px;" width="24"<br/>
alt="" height="24"></img></a>A <a href="/job/test_params/delete">Delete Project<<br/>
/a></div><div class="task"><a href="/job/test_params/configure"><img src="/stati<br/>
c/0d946710/images/24x24/setting.gif" style="margin: 2px;" width="24" alt="" heig<br/>
ht="24"></img></a>A <a href="/job/test_params/configure">Configure</a></div></di<br/>
v><table class="pane" id="buildHistory"><tr><td class="pane-header" colspan="2"><br/>
<div style="float:right">(<a href="/job/test_params/buildTimeTrend">trend</a>)</<br/>
div><div data="100" class="healthReport" onmouseover="this.className='healthRepo<br/>
rt hover';return true;<br/>
" onmouseout="this.className='healthReport';return true;"><a href="/job/<br/>
test_params/lastBuild"><img src="/static/0d946710/images/16x16/health-80plus.gif<br/>
" alt="100%"></img></a><div class="healthReportDetails"><table border="0"><thead<br/>
><tr><th align="left">W</th><th align="left">Description</th><th align="right">%<br/>
</th></tr></thead><tbody><tr><td align="left"><img src="/static/0d946710/images/<br/>
16x16/health-80plus.gif" title="" alt=""></img></td><td>Build stability: No rece<br/>
nt builds failed.</td><td align="right">100</td></tr></tbody></table></div></div<br/>
>Build History</td></tr><tr class="build-row "><td nowrap="nowrap"><img src="/st<br/>
atic/0d946710/images/16x16/blue.gif" width="16" tooltip="Success" alt="Success"<br/>
height="16"></img>A <br/>
      #3<br/>
    </td><td nowrap="nowrap"><a class="tip" href="/job/test_params/3/">Feb 18, 2<br/>
009 3:55:41 AM</a></td></tr><tr class="build-row "><td nowrap="nowrap"><img src=<br/>
"/static/0d946710/images/16x16/blue.gif" width="16" tooltip="Success" alt="Succe<br/>
ss" height="16"></img>A <br/>
      #2<br/>
    </td><td nowrap="nowrap"><a class="tip" href="/job/test_params/2/">Feb 18, 2<br/>
009 3:55:26 AM</a></td></tr><tr class="build-row "><td nowrap="nowrap"><img src=<br/>
"/static/0d946710/images/16x16/blue.gif" width="16" tooltip="Success" alt="Succe<br/>
ss" height="16"></img>A <br/>
      #1<br/>
    </td><td nowrap="nowrap"><a class="tip" href="/job/test_params/1/">Feb 18, 2<br/>
009 3:55:22 AM</a></td></tr><tr class="build-row"><td align="right" colspan="2"><br/>
<a href="/job/test_params/rssAll"><img src="/static/0d946710/images/atom.gif" bo<br/>
rder="0" alt="Feed"></img> for all</a>A <a href="/job/test_params/rssFailed"><im<br/>
g src="/static/0d946710/images/atom.gif" border="0" alt="Feed"></img> for failur<br/>
es</a></td></tr></table><script defer="true"><br/>
    updateBuildHistory("/job/test_params/buildHistory/ajax",4);<br/>
  </script></div></td><td width="80%" id="main-panel" height="100%"><a name="ski<br/>
p2content"></a><h1>Project test_params</h1><p>This build requires parameters:</p<br/>
><form action="build" method="post"><table width="100%"><tr><td class="setting-l<br/>
eftspace">A </td><td class="setting-name">TEST</td><td class="setting-main"><div<br/>
 name="parameter"><input type="hidden" value="TEST" name="name"></input><input n<br/>
ame="value" value="" type="text" class="setting-input "></input></div></td></tr><br/>
<tr class="validation-error-area"><td colspan="2"></td><td></td></tr><tr><td col<br/>
span="3"><input type="submit" class="submit-button" value="Build" name="Submit"><br/>
</input></td></tr></table></form></td></tr></table><table width="100%"><tr><td i<br/>
d="footer"><a href="http://hudson.dev.java.net/"></p>
<ul>
	<li>Closing connection #0<br/>
  Hudson ver. 1.285</a></td></tr></table></body></html></li>
</ul>

---
<details><summary><i>Originally reported by <a title="dbowler_cn" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dbowler_cn">dbowler_cn</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-3121" target="_blank">Hudson returns http 405 when parameterized option is checked</a></i></summary>
<i><ul>
<li><b>reported by</b>: dbowler_cn
<li><b>status</b>: Reopened
<li><b>priority</b>: Major
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 2
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=andrewr">andrewr</a>:</i>
<p>I think the sending of a 405 has to be a bug. I can't figure out from the<br/>
situation why this would be the case, so I had to look up what a 405 is used in.<br/>
From RFC 2616:</p>

<p>10.4.6 405 Method Not Allowed</p>

<p>The method specified in the Request-Line is not allowed for the resource<br/>
identified by the Request-URI. The response MUST include an Allow header<br/>
containing a list of valid methods for the requested resource. </p>


<p>We don't send back a Allow header and it just doesn't make sense to me.</p>

<p>The case is pretty rare, I did see it reported once on the Hudson mail list in<br/>
another context, someone trying to trigger builds thru a blackberry:<br/>
 <a href="http://www.nabble.com/triggering-builds-from-a-Blackberry-possible--td21100243.html" class="external-link" target="_blank" rel="nofollow noopener">http://www.nabble.com/triggering-builds-from-a-Blackberry-possible--td21100243.html</a><br/>
I think thats cause they're probably using a silently proxying browser.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=andrewr">andrewr</a>:</i>
<p>Tom--</p>

<p>I'm not sure I understand. A redirect (301 or 302) would be just fine.</p>

<p>But a 405 isn't a redirect, and it tells a reverse proxy to give up or at least<br/>
do something not predictable.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=qianqiao">qianqiao</a>:</i>
<p>I'm not sure using GET requests on /job/x/buildWithParameters is even a proper workaround, since it means that the "build" link on the left nav bar is completely useless, since it always points /job/x/build</p>

<p>The fact is that 405 will cause Jenkins' UI to be unusable on RPs that generate their own error page, and I don't think there's a valid reason to use 405 instead of 302 here.</p>

<p>I think this issue deserves to be re-opened and given more thought.</p>
